analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -690,15 +681,13 @@ public void TryCatchFinally_ThrowInCatch()
                 "InFinally",
                 "AfterFinally");
 
-            validator.TagStates("InCatch").Should().HaveCount(1)
-                .And.ContainSingle(x => HasUnknownException(x));
+            validator.TagStates("InCatch").Should().HaveCount(1).And.ContainSingleUnknownException();
```

Comment:
If something goes wrong, 
**before:**
```
Expected validator.TagStates("AfterFinally") to contain a single item matching HasUnknownException(x), but no such item was found.
```
**after:**
```
Expected given collection to contain a single state with unknown exception but no such state was found.
```
There's too much value lost, we're asserting 3 different states and it is not clear what went wrong :(

Reply:
Instead of specifying a method name, we mention exactly that the collection doesn't have a single state with an unknown exception.

Reply:
That's exactly the problem. Consider:
```
validator.TagStates("BeforeOuterTry").Should()....
validator.TagStates("BeforeInnerTry").Should()....
validator.TagStates("InInnerTry").Should()....
validator.TagStates("InInnerCatch").Should()....
validator.TagStates("InInnerFinally").Should()....
validator.TagStates("AfterInnerTry").Should()....
validator.TagStates("InOuterCatch").Should()....
validator.TagStates("InOuterFinally").Should()....
validator.TagStates("AfterOuterFinally").Should()....
```
Knowing what exactly failed is a must-have.

Reply:
The solution I've figured above solves this problem.